### ISSUE TITLE
Raise an exception if no SQLAlchemy session was set

### ISSUE THREAD

**Issue by lukasjuhrich** [FactoryBoy/factory_boy#281]
I just inherited from the wrong class, so no `session` attribute was set. Instead of a clear excetption, the code failed implicitly when the object was tried to be added to the session.

It would be a nice convenience feature to raise a ValueError or similiar. 

Proof of concept:

``` python
from factory.alchemy import SQLAlchemyModelFactory as Factory
from factory.fuzzy import FuzzyText
from flask.ext.sqlalchemy import SQLAlchemy
from sqlalchemy import (Column, String)

db = SQLAlchemy()


class Traffic(db.Model):
    traffic_id = Column(String(15), nullable=False, index=True, primary_key=True)


class TrafficFactory(Factory):
    class Meta:
        model = Traffic
        # this was missing:
        # session = db.session

    traffic_id = FuzzyText()


if __name__ == '__main__':
    assert TrafficFactory()

```

Returns: 

```
AttributeError: 'NoneType' object has no attribute 'add'
```

The place where it fails is alchemy.py:46.
Adding  `if not session: raise ValueError("No session provided")` before that line should be sufficient.

**Comment by jeffwidman** [comment]
Good idea. :+1: 

Two potential edge cases:

1. I'm not 100% sure, but I don't think you need a `session` for the `build` strategy, only the `create` strategy, as there are times you might want to create an object and only then decide which session to pass it to. Again, I'm not 100% sure on this, I just want to be careful to not prevent creative uses of `build` or `stub`.
2. `sessions` can be inherited from a base factory class that specifies the `session`, they don't need to be manually specified for every class. So any check for `session` needs to be checking all the way up the inheritance chain.

Can you put together a PR with this? 
Please also include a basic test verifying the exception is thrown correctly.

**Comment by rbarrois** [comment]
Just a note here: the session inheritance is handled in ``factory.base.FactoryOptions``.

For this case, it might be simpler to put the check with the ``SQLAlchemyModelFactory._create`` code, just before accessing the session.

This ways, you don't have to worry about the way sessions are passed across the stack, nor handling the ``abstract=True`` factories in that chain.

**Comment by lukasjuhrich** [comment]
I don't have too much knowledge about the internals of SQLAlchemy and factory_boy, but I will gladly try to do so.

**Comment by rbarrois** [comment]
@lukasjuhrich have you had a chance to look into it?

I think the best place would be around here: https://github.com/rbarrois/factory_boy/blob/master/factory/alchemy.py#L44

You could then check for ``session is None``, and provide a helpful error message ;)

**Comment by lukasjuhrich** [comment]
@rbarrois I didn't expect to be that busy the last weeks. Thanks for pointing me in the right direction, anyway!

**Comment by jeffwidman** [comment]
Fixed by #339
